File: arrays-and-strings/121. Best Time to Buy and Sell Stocks/solution.ts
function maxProfit(prices: number[]): number {
  let minPrice = 10e9;
  let maxProfit = 0;
  let profit = 0;

  for (const p of prices) {
    if (p < minPrice) minPrice = p;
    profit = p - minPrice;
    if (profit > maxProfit) maxProfit = profit;
  }
  return maxProfit;
}

export { maxProfit };

File: arrays-and-strings/121. Best Time to Buy and Sell Stocks/solution.py
class Solution:
    def maxProfit(self, prices: list[int]) -> int:
        if(len(prices) < 2):
            return 0

        profit = 0
        maxProfit = 0
        minPrice = 10e6

        for p in prices:
            if(p < minPrice):
                minPrice = p

            profit = p - minPrice

            if(profit > maxProfit):
                maxProfit = profit

        return maxProfit